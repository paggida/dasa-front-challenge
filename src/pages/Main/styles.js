import styled from "styled-components";
import metrics from "../../styles/metrics";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: ${metrics.basePadding * 3 + metrics.base};
  img {
    width: 250px;
    margin-bottom: ${metrics.baseMargin * 3 + metrics.base};
  }
`;
