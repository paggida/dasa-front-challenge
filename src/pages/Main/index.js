import React from "react";
import { Container } from "./styles";
import logo from "../../assets/logo.png";
import UserForm from "../../components/UserForm";

const Main = () => (
  <Container>
    <img src={logo} alt="DasaHub" />
    <UserForm />
  </Container>
);

export default Main;
