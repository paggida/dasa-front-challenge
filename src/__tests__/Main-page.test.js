import React from 'react';
import Main from '../pages/Main';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json"


it("Should render the homepage", () => {
  const page = mount(<Main />);
  expect(toJson(page)).toMatchSnapshot();
  page.unmount();
});
