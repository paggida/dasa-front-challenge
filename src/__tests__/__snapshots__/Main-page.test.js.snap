// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the homepage 1`] = `
<Main>
  <styled.div>
    <div
      className="sc-AxjAm dHUTpM"
    >
      <img
        alt="DasaHub"
        src="logo.png"
      />
      <UserForm
        handleOutput={[Function]}
      >
        <styled.form
          onSubmit={[Function]}
          withError={false}
        >
          <form
            className="sc-AxirZ lbOMdt"
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Usuário do GitHub"
              type="text"
              value=""
            />
            <button
              id="searchBtn"
              type="submit"
            >
              Buscar repositórios
            </button>
          </form>
        </styled.form>
      </UserForm>
      <RepositoriesList
        repositories={Array []}
      >
        <div />
      </RepositoriesList>
    </div>
  </styled.div>
</Main>
`;
